### PR TITLE
ci: remove `protoc` installation

### LINKED ISSUE #3539: Remove `protoc` from CI workflow
## Description

With https://github.com/libp2p/rust-libp2p/issues/3024 being merged, we will eventually be able to remove `protoc` from our CI. However, we need to do another release first because `cargo semver-checks` still needs to be able to build the previous release to do the rustdoc JSON diffing and as of now, the last release still requires protoc.

## Requirements

1.
2.
3.

## Are you planning to do it yourself in a pull request?

Yes / No / Maybe.

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

With all crates have received a release since https://github.com/libp2p/rust-libp2p/issues/3024, building the baseline rustdoc no longer required `protoc` and we can thus remove it from our CI entirely.

Resolves #3539.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by thomaseizinger** [comment]
I've opened https://github.com/paritytech/soketto/issues/72 which might allow us to remove `zlib`.

File: transports/websocket/src/lib.rs [hunk 1]
```diff
@@ -52,6 +52,10 @@ use std::{
 ///
 /// If you don't need Secure Websocket's support, use a plain TCP transport as an inner transport.
 ///
+/// # Dependencies
+///
+/// This transport requires the `zlib` shared library to be installed on the system.
```
Comment (mxinden) [inline transports/websocket/src/lib.rs:57]: Let's link to https://github.com/paritytech/soketto/issues/72 here?

**Review by mxinden — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,8 +29,7 @@ jobs:
       matrix:
         crate: ${{ fromJSON(needs.gather_published_crates.outputs.members) }}
     steps:
-      - name: Install Protoc
-        run: sudo apt-get install -y protobuf-compiler
+      - run: sudo apt-get install zlib1g-dev --yes
```
Comment (mxinden) [inline .github/workflows/ci.yml]: Also let's add a comment here for future us.
```suggestion
     # Needed for `libp2p-websocket`, more specifically `soketto`, see https://github.com/paritytech/soketto/issues/72.
      - run: sudo apt-get install zlib1g-dev --yes
```

**Comment by mxinden** [comment]
Thanks for debugging this!

File: transports/websocket/src/lib.rs (hunk 1 above)
Comment (thomaseizinger) [inline transports/websocket/src/lib.rs:57]:
```suggestion
/// This transport requires the `zlib` shared library to be installed on the system.
/// 
/// Future releases might lift this requirement, see <https://github.com/paritytech/soketto/issues/72>.
```

File: .github/workflows/ci.yml (hunk 2 above)
Comment (thomaseizinger) [inline .github/workflows/ci.yml]: Also let's add a comment here for future us.
```suggestion
      - name: Install zlib for `libp2p-websocket` # https://github.com/paritytech/soketto/issues/72
        run: sudo apt-get install zlib1g-dev --yes
```

**Comment by drHuangMHT** [comment]
@mxinden Approving review required to automatically merge this.

**Review by mxinden — APPROVED** [review]
Thanks for the latest additions.

**Pull request merged**